compose/snippets/src/main/java/com/example/compose/snippets/predictiveback/PredictiveBackSnippets.kt

```diff
@@ -0,0 +1,190 @@
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.material3.Surface
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableFloatStateOf
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.TransformOrigin
+import androidx.compose.ui.input.pointer.util.VelocityTracker
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.unit.dp
+import androidx.navigation.NavHostController
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import androidx.navigation.compose.rememberNavController
+import kotlin.coroutines.cancellation.CancellationException
+import kotlinx.coroutines.flow.Flow
+
+@Composable
+private fun PredictiveBackOverrideExit(
+    modifier: Modifier,
+) {
+    val navController = rememberNavController()
+
+    // [START android_compose_predictiveback_navhost]
+    NavHost(
+        navController = navController,
+        startDestination = "home",
+        popExitTransition = {
+            scaleOut(
+                targetScale = 0.9f,
+                transformOrigin = TransformOrigin(pivotFractionX = 0.5f, pivotFractionY = 0.5f)
+            )
+        },
+        popEnterTransition = {
+            EnterTransition.None
+        },
+        modifier = modifier,
+    )
+    // [END android_compose_predictiveback_navhost]
+    {
+        composable("home") {
+            HomeScreen(
+                modifier = modifier,
+                navController = navController,
+            )
+        }
+        composable("settings") {
+            SettingsScreen(
+                modifier = modifier,
+                navController = navController,
+            )
+        }
+    }
+}
+
+@Composable
+private fun HomeScreen(
+    modifier: Modifier = Modifier,
+    navController: NavHostController
+) {
+}
+
+@Composable
+private fun SettingsScreen(
+    modifier: Modifier = Modifier,
+    navController: NavHostController
+) {
+}
+
+@Composable
+private fun PredictiveBackHandlerBasicExample() {
+
+    var boxScale by remember { mutableFloatStateOf(1F) }
+
+    Box(
+        modifier = Modifier
+            .fillMaxSize(boxScale)
+            .background(Color.Blue)
+    )
+
+    // [START android_compose_predictivebackhandler_basic]
+    PredictiveBackHandler(true) { progress: Flow<BackEventCompat> ->
+        // code for gesture back started
+        try {
+            progress.collect { backEvent ->
+                // code for progress
+                boxScale = 1F - (1F * backEvent.progress)
+            }
+            // code for completion
```

Comment:
Shouldn't this have a completion state? From the looks of it, it should be `boxScale = 0F`

Reply:
Yep, it should have a completion state. Added it into the https://github.com/android/snippets/pull/399 PR.